apps/dapp-console-api/src/routes/rebates/RebatesRoute.ts

```diff
@@ -33,7 +56,293 @@ export class RebatesRoute extends Route {
+          },
+          'error fetching contract',
+        )
+        throw Trpc.handleStatus(500, 'error fetching contract')
+      })
+
+      if (!contract) {
+        throw Trpc.handleStatus(400, 'contract does not exist')
+      }
+
+      if (contract.state !== ContractState.VERIFIED) {
+        throw Trpc.handleStatus(403, 'cannot claim on unverified contract')
+      }
+
+      if (!isContractDeploymentDateEligibleForRebate(contract)) {
+        throw Trpc.handleStatus(
+          403,
+          'contract must be deployed after user signed up for dev console',
+        )
+      }
+
+      const rebateAlreadyClaimed = await hasContractAlreadyBeenClaimed({
+        db: this.trpc.database,
+        contractId,
+        chainId: contract.chainId,
+        contractAddress: contract.contractAddress,
+      }).catch((err) => {
+        metrics.fetchDeploymentRebateStatusErrorCount.inc()
+        this.logger?.error(
+          {
+            error: err,
+            entityId: user.entityId,
+            contractId: contractId,
+          },
+          'error fetching deployment rebate status',
+        )
+        throw Trpc.handleStatus(500, 'error fetching deployment rebate status')
+      })
+
+      if (rebateAlreadyClaimed) {
+        throw Trpc.handleStatus(
+          403,
+          'a rebate has already been claimed for this contract',
+        )
+      }
+
+      const cbVerifiedWallets = (
+        await getWalletVerifications({
+          db: this.trpc.database,
+          entityId: user.entityId,
+        }).catch((err) => {
+          metrics.fetchWalletVerificationsErrorCount.inc()
+          this.logger?.error(
+            {
+              error: err,
+              entityId: user.entityId,
+              contractId,
+            },
+            'error fetching verifications for wallet',
+          )
+          throw Trpc.handleStatus(
+            500,
+            'error fetching verifications for wallet',
+          )
+        })
+      ).cbVerifiedWallets
+
+      if (cbVerifiedWallets.length === 0) {
+        throw Trpc.handleStatus(
+          403,
+          'must have a cb verified wallet in order to claim rebate',
+        )
+      }
+
+      const totalAmountAlreadyClaimed = await getTotalRebatesClaimed({
+        db: this.trpc.database,
+        entityId: user.entityId,
+      }).catch((err) => {
+        metrics.fetchTotalRebatesClaimedErrorCount.inc()
+        this.logger?.error(
+          {
+            error: err,
+            entityId: user.entityId,
+          },
+          'error fetching total rebates claimed from db',
+        )
+        throw Trpc.handleStatus(500, 'error fetching total rebates claimed')
+      })
+
+      if (totalAmountAlreadyClaimed >= MAX_REBATE_AMOUNT) {
+        throw Trpc.handleStatus(
+          403,
+          'user has already claimed the max amount of rebates',
+        )
+      }
+
+      const amountLeftToClaim = MAX_REBATE_AMOUNT - totalAmountAlreadyClaimed
```

Comment:
might be good to force this to 0 instead of it possibly being able to go negative. i can see us having to potentially deal with some level of inaccurary where the claimed amount might be ever so slighly higher than the max rebate amount